src/main/java/eutros/multiblocktweaker/helper/ReflectionHelper.java

```diff
@@ -1,50 +1,57 @@
-                    .invoke(object);
-        } catch (Error | RuntimeException e) {
-            throw e;
-        } catch (Throwable e) {
-            throw new RuntimeException(e);
-        }
+        return (T) handles
+                .computeIfAbsent(fieldClass, c -> new HashMap<>())
+                .computeIfAbsent(fieldName, computeHandle(fieldClass, fieldName))
+                .map(f-> {
+                    try {
+                        return f.get(object);
+                    } catch (IllegalAccessException e) {
+                        CraftTweakerAPI.logError(String.format("No static field \"%s\" found", fieldName));
+                    }
+                    return null;
+                }).orElse(null);
     }
 
-    private static <C> Function<String, MethodHandle> computeHandle(Class<? super C> fieldClass, String fieldName) {
+    @SuppressWarnings("unchecked")
+    public static <T, C> T getStatic(Class<? super C> fieldClass, String fieldName) throws ClassCastException {
+        return (T) handles
+                .computeIfAbsent(fieldClass, c -> new HashMap<>())
+                .computeIfAbsent(fieldName, computeHandle(fieldClass, fieldName))
+                .map(f-> {
+                    try {
+                        return f.get(fieldClass);
+                    } catch (IllegalAccessException e) {
+                        CraftTweakerAPI.logError(String.format("No static field \"%s\" found", fieldName));
+                    }
+                    return null;
+                }).orElse(null);
+    }
+
+    private static <C> Function<String, Optional<Field>> computeHandle(Class<? super C> fieldClass, String fieldName) {
         return p -> {
             try {
-                return MethodHandles.publicLookup()
-                        .unreflectGetter(
-                                Preconditions.checkNotNull(
-                                        FieldUtils.getField(
-                                                fieldClass,
-                                                fieldName,
-                                                true
-                                        ),
-                                        "Couldn't find field %s of %s.",
-                                        fieldName, fieldClass
-                                )
-                        );
-            } catch (IllegalAccessException e) {
-                throw new IllegalStateException(String.format("Couldn't access field %s of %s", fieldName, fieldClass), e);
+                Field field = fieldClass.getDeclaredField(fieldName);
+                field.setAccessible(true);
+                return Optional.of(field);
+            } catch (NoSuchFieldException e) {
+                CraftTweakerAPI.logError(String.format("Couldn't access field %s of %s", fieldName, fieldClass), e);
+                return Optional.empty();
             }
         };
     }
```

Comment:
Why :(

Reply:
the privous method cant get the static value of fields.

Reply:
`getStatic` is just
```java
    @SuppressWarnings("unchecked")
    public static <T> T getStatic(Class<?> fieldClass, String fieldName) throws ClassCastException {
        try {
            return (T) handles
                    .computeIfAbsent(fieldClass, c -> new HashMap<>())
                    .computeIfAbsent(fieldName, computeHandle(fieldClass, fieldName))
                    .invoke();
        } catch (Error | RuntimeException e) {
            throw e;
        } catch (Throwable e) {
            throw new RuntimeException(e);
        }
    }
```
so could you use that instead of rewriting the whole class, please?